### DISCUSSION TITLE
Protocols

### DISCUSSION THREAD

**Discussion by UniqueName12345** [CatalystDevOrg/Catalyst#321] [Proposals]
I propose to bring back the exec:// protocol, but add a confirmation warning like with the steam:// protocol.

If you say "no", then it won't execute. If you say "yes", then it will execute in a sandboxed environment.

Oh, and add these protocols as well:

- explorer:// - Opens your file explorer
- shell://[command] - Runs a command in the terminal

**Comment by jdev082** [comment]
> I propose to bring back the exec:// protocol, but add a confirmation warning like with the steam:// protocol.
> 
> If you say "no", then it won't execute. If you say "yes", then it will execute in a sandboxed environment.
> 
> Oh, and add these protocols as well:
> 
>     * explorer:// - Opens your file explorer
> 
>     * shell://[command] - Runs a command in the terminal

Both are impossible. As the underlying UX (what's being rendered by Electron) has no access to NodeJS. Meaning, they cannot execute commands. If the JS within the UX could somehow execute system commands this could be possible. But that's yet to be determined.

**Reply by redstone-dev** [reply]
It is...

**Reply by jdev082** [reply]
Due to recent discovery of resources this feature will be reconsidered.

**Comment by jdev082** [comment]
@UniqueName12345 This feature will be implemented but not in the exact way you've mentioned. Instead, all the protocols registered by the system will be supported. Various programs register a protocol that allows them to be interacted with via browser URLs.

**Comment by jdev082** [comment]
This will be reconsidered
